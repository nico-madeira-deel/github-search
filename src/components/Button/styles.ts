import styled, { css } from 'styled-components'

export const Wrapper = styled.button`
  border: none;
  cursor: pointer;

  ${({ theme }) => css`
    background-color: ${theme.colors.pink};
    border-radius: ${theme.border.radius};
    color: ${theme.colors.darkBlue};
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.small};
    transform: scale(1);
    transition: transform ${theme.transitions.fast};

    &:hover,
    &:focus {
      transform: scale(1.1);
      transition: transform ${theme.transitions.fast};
    }

    &:disabled {
      cursor: not-allowed;
      opacity: 0.5;
      transform: scale(1);
    }
  `}
`
