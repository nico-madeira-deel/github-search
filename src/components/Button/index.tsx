import React from 'react'
import * as S from './styles'

export type ButtonProps = {
  children: React.ReactNode
  disabled?: boolean
  onClick: () => void
}

const Button = ({ children, disabled = false, onClick }: ButtonProps) => (
  <S.Wrapper disabled={disabled} onClick={onClick} type="button">
    {children}
  </S.Wrapper>
)

export default Button
