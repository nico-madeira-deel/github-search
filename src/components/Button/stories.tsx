import { Story, Meta } from '@storybook/react/types-6-0'
import Button, { ButtonProps } from '.'

export default {
  argTypes: {
    onClick: {
      type: 'function'
    }
  },
  title: 'Button',
  component: Button
} as Meta

export const Basic: Story<ButtonProps> = (args) => (
  <Button {...args}>Test</Button>
)

Basic.args = {
  onClick: () => console.log('test')
}
