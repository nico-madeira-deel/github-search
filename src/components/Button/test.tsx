import { screen, fireEvent } from '@testing-library/react'

import Button from '.'
import { renderWithTheme } from 'utils/tests/helpers'

const initialProps = {
  onClick: jest.fn()
}

describe('<Button />', () => {
  it('should render the button', () => {
    const { container } = renderWithTheme(
      <Button {...initialProps}>Button</Button>
    )

    expect(screen.getByRole('button', { name: /button/i })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should have disabled attribute when prop disabled its true', () => {
    renderWithTheme(
      <Button {...initialProps} disabled={true}>
        Button disabled
      </Button>
    )

    expect(
      screen.getByRole('button', { name: /button disabled/i })
    ).toBeDisabled()
  })

  it('should call the onclick when button its clicked', () => {
    renderWithTheme(<Button {...initialProps}>Button clicked</Button>)
    const button = screen.getByRole('button', { name: /button clicked/i })

    fireEvent.click(button, initialProps.onClick)

    expect(initialProps.onClick).toHaveBeenCalledTimes(1)
  })
})
