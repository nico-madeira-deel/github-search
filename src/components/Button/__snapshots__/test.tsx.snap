// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
.c0 {
  border: none;
  cursor: pointer;
  outline: none;
  background-color: #ffd6ba;
  border-radius: 0.5rem;
  color: #555b6e;
  font-size: 1.6rem;
  font-weight: 700;
  padding: 0.8rem 2.4rem;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
}

.c0:hover,
.c0:focus {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.5;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<button
  class="c0"
  type="button"
>
  Button
</button>
`
